### DISCUSSION TITLE
Using Paste for customer-facing demo app

### DISCUSSION THREAD

**Discussion by waijeeho** [twilio-labs/paste#1590] [Q&A] [answered]
Hi team,

I'm beginning work on a web/Javascript demo app with the Conversations SDK team – [screenshot of the current state of the codesandbox demo is here](https://www.figma.com/file/rZBWxXEUckeFXQrUeSRUms/Conversations-SDK-Demo?node-id=781%3A0). We would like to use the Paste system for the customer-facing web demo. As we plan our work, I'd love to get your guidance and advice:

- Has Paste been used for customer-facing demos before? 
- Is this a reasonable approach given that customers will have access to the demo app code? Are there things we should watch out for?
- In addition, there will be some communication-specific UI that we'll build to support the Conversations experience that I believe does not exist in the Paste system today (eg, message history, reactions). Is it alright to build custom UI and pair it with Paste?

Thanks for your help!
Wai-Jee Ho

**Comment by TheSisb** [comment, marked as answer]
Hi @waijeeho,

> Has Paste been used for customer-facing demos before?

Yup! Paste has been used in customer-facing demos, Twilio's hack week (Tweek), customer-facing products, and even by customers themselves as of recently to build UIs in Flex.  🎉 

> Is this a reasonable approach given that customers will have access to the demo app code? Are there things we should watch out for?

Good question.  Since Paste is fully open-source and public, we make sure [not to include any sensitive information into our components](https://paste.twilio.design/getting-started/about-paste#paste-is-not). In fact, our components are purely [presentational](https://learn.co/lessons/react-presentation-components) &mdash; this means they have no business logic or proprietary information within them. As long as you don't add any sensitive information in the code you are sharing, you can rest easy knowing Paste is safe to use.

> In addition, there will be some communication-specific UI that we'll build to support the Conversations experience that I believe does not exist in the Paste system today (eg, message history, reactions). Is it alright to build custom UI and pair it with Paste?

Another excellent question :). Because we are a small team supporting all of Twilio and Twilio customers, we knew we wouldn't be able to build everything people wanted directly into Paste. That's why Paste was [designed to be composable](https://paste.twilio.design/tokens/how-to-compose-custom-ui-with-tokens).  By using a combination of our [primitive components](https://paste.twilio.design/getting-started/about-paste#glossary-of-terms) (like [Box](https://paste.twilio.design/primitives/box) and [Text](https://paste.twilio.design/primitives/text)) along with our [tokens](https://paste.twilio.design/tokens), you should be able to create any custom UI your product needs in a consistent way with the rest of Twilio's branding.

Hope this helps answer your questions. Feel free to reach back out if anything is still unclear!

**Reply by waijeeho** [reply]
Thank you @TheSisb!
